File: app/src/test/java/com/flatstack/android/utils/storage/RuntimeStorage.java
package com.flatstack.android.utils.storage;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class RuntimeStorage implements IStorage {

    private Map<String, Object> map = new HashMap<>();

    @Override public <T> void put(@NonNull String key, @NonNull T item) {
        map.put(key, item);
    }

    @Nullable @Override
    public <T> T get(@NonNull String key, @NonNull Type clazz) {
        return (T) map.get(key);
    }

    @Override public void putLong(@NonNull String key, long number) {
        map.put(key, Long.valueOf(number));
    }

    @Override public long getLong(@NonNull String key, long defaultValue) {
        Object o = map.get(key);
        return o != null ? (long) o : defaultValue;
    }

    @Override public void putInt(@NonNull String key, int number) {
        map.put(key, Integer.valueOf(number));
    }

    @Override public int getInt(@NonNull String key, int defaultValue) {
        Object o = map.get(key);
        return o != null ? (int) o : defaultValue;
    }

    @Override public void putBoolean(@NonNull String key, boolean value) {
        map.put(key, Boolean.valueOf(value));
    }

    @Override public boolean getBoolean(@NonNull String key, boolean defaultValue) {
        Object o = map.get(key);
        return o != null ? (boolean) o : defaultValue;
    }

    @Override public void putString(@NonNull String key, @NonNull String str) {
        map.put(key, str);
    }

    @Override public String getString(@NonNull String key) {
        Object o = map.get(key);
        return (o != null) ? (String) o : null;
    }

    @Override public void remove(@NonNull String key) {
        map.remove(key);
    }

    @Override public <T> void putCollection(@NonNull String key, @NonNull List<T> items) {
        map.put(key, items);
    }

    @Override public <T> List<T> getCollection(@NonNull String key, @NonNull Type type) {
        return (List<T>) map.get(key);
    }
}

File: app/src/test/java/com/flatstack/android/utils/storage/RuntimeStorage.kt
package com.flatstack.android.utils.storage

import com.flatstack.android.util.storage.IStorage
import java.lang.reflect.Type
import java.util.*

class RuntimeStorage : IStorage {

    private val map = HashMap<String, Any>()

    override fun <T> get(key: String, clazz: Type): T? {
        return map[key] as T?
    }

    override fun putLong(key: String, number: Long) {
        map[key] = java.lang.Long.valueOf(number)
    }

    override fun getLong(key: String, defaultValue: Long): Long {
        val o = map[key]
        return if (o != null) o as Long else defaultValue
    }

    override fun putInt(key: String, number: Int) {
        map[key] = Integer.valueOf(number)
    }

    override fun getInt(key: String, defaultValue: Int): Int {
        val o = map[key]
        return if (o != null) o as Int else defaultValue
    }

    override fun putBoolean(key: String, value: Boolean) {
        map[key] = java.lang.Boolean.valueOf(value)
    }

    override fun getBoolean(key: String, defaultValue: Boolean): Boolean {
        val o = map[key]
        return if (o != null) o as Boolean else defaultValue
    }

    override fun putString(key: String, str: String) {
        map[key] = str
    }

    override fun getString(key: String): String? {
        val o = map[key]
        return if (o != null) o as String? else null
    }

    override fun remove(key: String) {
        map.remove(key)
    }

    override fun <T> getCollection(key: String, type: Type): List<T>? {
        return map[key] as List<T>?
    }

    override fun put(key: String, items: Any) {
        map[key] = items
    }

    override fun <T> putCollection(key: String, items: List<T>) {
        map[key] = items
    }
}
